pkg/tfvars/vsphere/vsphere.go

```diff
@@ -56,6 +60,22 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 	// /<datacenter>/vm/<folder_path> so we can split on "vm/".
 	folderRelPath := strings.SplitAfterN(controlPlaneConfig.Workspace.Folder, "vm/", 2)[1]
 
+	// Must use the Managed Object ID for a port group (e.g. dvportgroup-5258)
+	// instead of the name since port group names aren't always unique in vSphere.
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1918005
+	vim25Client, _, err := vsphere.CreateVSphereClients(context.TODO(),
+		controlPlaneConfig.Workspace.Server,
+		sources.Username,
+		sources.Password)
+	networkID, err := vsphere.GetNetworkMoID(context.TODO(),
+		vim25Client,
+		controlPlaneConfig.Workspace.Datacenter,
+		sources.Cluster,
+		controlPlaneConfig.Network.Devices[0].NetworkName)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get vSphere network ID")
+	}
```

Comment:
Let's find the networkID in `tfvars` in the `asset` pkg and pass it in here in `TFVarsSources`.

Reply:
Done.